packages/flutter/test/semantics/semantics_update_test.dart

```diff
@@ -0,0 +1,188 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:typed_data';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('Semantics update', () async {
+    final SemanticsUpdateTestBinding binding = SemanticsUpdateTestBinding();
+    binding.pipelineOwner.ensureSemantics();
+    await binding.runTest(() async {
+      binding.attachRootWidget(
+        MaterialApp(
+          home: MergeSemantics(
+            child: Semantics(
+              value: 'test 1',
+              textField: true,
+              // This semantics b
+              child: MergeSemantics(
+                child: Semantics(
+                  value: 'test 2',
+                  textField: true,
+                  child: const Text('test 3'),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+      SemanticsUpdateBuilderSpy.observations.clear();
+      binding.scheduleFrame();
+      await binding.pump(null, EnginePhase.sendSemanticsUpdate);
+
+      // Checks all nodes are connected.
+      // Starts with root node.
+      int currentNode = 0;
+      while(SemanticsUpdateBuilderSpy.observations.containsKey(currentNode)) {
+        final SemanticsNodeUpdateObservation observation =  SemanticsUpdateBuilderSpy.observations.remove(currentNode)!;
+        if (observation.childrenInTraversalOrder.isEmpty)
+          break;
+        expect(observation.childrenInTraversalOrder.length, 1);
+        currentNode = observation.childrenInTraversalOrder[0];
+      }
+      // We should have looped through the all the observations.
+      expect(SemanticsUpdateBuilderSpy.observations.isEmpty, isTrue);
+    }, () { });
+
+    SemanticsUpdateBuilderSpy.observations.clear();
+  });
+
+}
```

Comment:
There should also be a test where we modify the semantics of a node that gets merged into another node and ensure that only the (updated) merged node is send over.